### PR TITLE
Add feature with s3 state store from configfile

### PR DESCRIPTION
**Pull request by jsenon**
Add state store config file on kops command line. #2435

### CODE REVIEW

**Comment by jsenon** [comment]
@mikesplain I ve updated my previous proposal. could you approve ?

**Comment by jsenon** [comment]
@chrislovecnm  could you approve?

File: cmd/kops/util/factory.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ Trailing slash will be trimmed.`
 func (f *Factory) Clientset() (simple.Clientset, error) {
 	if f.clientset == nil {
 		registryPath := f.options.RegistryPath
+		fmt.Println("Config S3:", registryPath)
```
[the remark this answers is not in the document]
Comment (jsenon) [reply cmd/kops/util/factory.go]: I will remove, but I keep in mind to introduce verbosity

File: cmd/kops/root.go [hunk 2]
```diff
@@ -150,18 +148,45 @@ func NewCmdRoot(f *util.Factory, out io.Writer) *cobra.Command {
...
+		viper.AutomaticEnv()
+	}
+
 	if rootCommand.configFile != "" {
 		// enable ability to specify config file via flag
 		viper.SetConfigFile(rootCommand.configFile)
+		if err := viper.ReadInConfig(); err == nil {
+			viper.BindEnv("KOPS_STATE_STORE")
+			rootCommand.RegistryPath = viper.GetString("KOPS_STATE_STORE")
+			fmt.Println("Read Default config File, s3 :", rootCommand.RegistryPath)
```
[the remark this answers is not in the document]
Comment (jsenon) [reply cmd/kops/root.go]: Indeed, I will remove also

**Comment by justinsb** [comment]
I love the feature, but it feels like the code has duplication (do we have to call BindEnv 3 times, for example?). I guess I need to better understand the viper API to figure out what's going on here...

Also it wasn't clear what the format of the conf file was (though this is primarily because I haven't yet dug into the viper API).  An example would be great :-)  In particular though, I'm wondering if the keys in the conf file looks like `KOPS_STATE_STORE` or look like `config` - I think I'd prefer the second, personally, because that's closer to the flags.

**Comment by jsenon** [comment]
@justinsb, do I need to improve something in my pr ?

**Comment by jsenon** [comment]
@chrislovecnm I don't understand issue regarding CLA, have you some ideas?

**Comment by justinsb** [comment]
/approve
/lgtm

Thanks @jsenon

**Comment by justinsb** [comment]
Fixed the merge conflicts; reapplying lgtm

/lgtm

**Comment by justinsb** [comment]
My bad, so going to force-merge & fixup whatever gofmt errors I introduced

**Comment by jsenon** [comment]
Thanks for merge

**Pull request merged**